### PR TITLE
Update logger

### PR DESCRIPTION
**Pull request by jverswijver**
small changes to logging behavior:
- add check to update the logger when the `dj.config` is updated
- make sure that environment variables are given priority over the config
- should work with context manager now

### CODE REVIEW

**Comment by guzman-raphael** [comment]
@jverswijver From our prior discussion, the largest issue I see here is that if we are properly addition `loglevel` to be used in the config, we should verify the case where a user has a `dj_local_conf.json` and first imports `datajoint`. Their log level should be appropriately set.

**Comment by jverswijver** [comment]
Yes, we should try to look for the config and if it is there we should load the log level for the user when we create the logger. Except of they have the log level set in their env which should take priority.

File: datajoint/settings.py [hunk 1]
```diff
@@ -248,6 +248,9 @@ def __setitem__(self, key, value):
                 self._conf[key] = value
             else:
                 raise DataJointError("Validator for {0:s} did not pass".format(key))
+            valid_logging_levels = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
+            if "loglevel" in key and value in valid_logging_levels:
```
Comment (dimitri-yatsenko) [inline datajoint/settings.py:252]:
```suggestion
            if "loglevel" == key and value in valid_logging_levels:
```

Comment (dimitri-yatsenko) [reply datajoint/settings.py:252]: should probably raise an error otherwise?

**Comment by dimitri-yatsenko** [comment]
These changes are in implemented in #1186.

**Pull request closed without merging**